### ISSUE TITLE
CRD validation require deprecated fields

### ISSUE THREAD

**Issue by ese** [coreos/prometheus-operator#1043]
**What did you do?**

Test prometheus-operator 0.17 with our current prometheus definition

**What did you expect to see?**

Works fine

**What did you see instead? Under which circumstances?**

Failed to validate storage spec
spec.storage.class in body is required
spec.storage.selector in body is required
spec.storage.resources in body is required

If I deactivate validation all works fine with the same spec

**Environment**

* Kubernetes version information:

1.9.3

* Kubernetes cluster kind: 

kops

* Manifests:

```
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  namespace: monitoring
  name: k8s
  labels:
    prometheus: k8s
    kubernetes.io/cluster-service: "true"
spec:
  version: 2.1.0
  serviceAccountName: prometheus-k8s
  serviceMonitorSelector:
    matchLabels:
      prometheus: k8s
  ruleSelector:
    matchLabels:
      role: prometheus-rulefiles
      prometheus: k8s
  evaluationInterval: 15s
  alerting:
    alertmanagers:
    - namespace: monitoring
      name: alertmanager-main
      port: web
  storage:
    volumeClaimTemplate:
      spec:
        resources:
          requests:
            storage: 50G
```

**Comment by brancz** [comment]
This is already fixed on master, we just have to do a release, sorry for the inconvenience.

**Comment by brancz** [comment]
There is one more feature I would like to implement then we'll do another release.

**Comment by bgagnon** [comment]
I also experienced this after upgrading a little too hastily to 0.17.
This redundant block helped pass validation:

```yaml
    # FIXME: CRD validation bug -- deprecated fields
    # https://github.com/coreos/prometheus-operator/issues/992
    class: ""
    selector: ""
    resources:
      requests: {}
```
Sticking with `v0.16.1` until this is fixed.

**Comment by brancz** [comment]
@incognick currently validations are not updated, so you will need to create the CRDs from scratch or apply them from `example/prometheus-operator-crd`.

**Comment by FaKod** [comment]
@brancz Does 0.17 still use the "wrong" CRDs? 
0.18 works for me (no storage validation errors)

**Comment by brancz** [comment]
Yep, this can be closed as this problem went away with 0.18. Thanks!
